### PR TITLE
feat(NcDialogButton): Allow to return `false` from callback to keep dialog open

### PR DESCRIPTION
**Pull request by susnux**
### ☑️ Resolves

If the callback returns `false` then the click event will not be forwarded. This could be useful if the click triggers a validation that fails and the user should be able to fix the issue.

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [x] 📘 Component documentation has been extended, updated or is not applicable
- [x] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

File: src/components/NcDialogButton/NcDialogButton.vue [hunk 1]
```diff
@@ -103,9 +104,10 @@ export default defineComponent({
 		 * Handle clicking the button
 		 * @param {MouseEvent} e The click event
 		 */
-		const handleClick = (e) => {
-			props.callback?.()
-			emit('click', e)
+		const handleClick = async (e) => {
```
Comment (st3iny) [inline src/components/NcDialogButton/NcDialogButton.vue:107]: Why is this async?

**Review by st3iny — APPROVED** [review]
Code looks good otherwise.

File: src/components/NcDialogButton/NcDialogButton.vue (hunk 1 above)
Comment (ShGKme) [reply src/components/NcDialogButton/NcDialogButton.vue:107]: I think, we can call callback with await (and set button to loading)

Comment (susnux) [reply src/components/NcDialogButton/NcDialogButton.vue:107]: Yes that was the idea, I am not sure if I should also include it here or follow up

Comment (ShGKme) [reply src/components/NcDialogButton/NcDialogButton.vue:107]: > include it here

Yes plz ༼ つ ◕_◕ ༽つ

Comment (susnux) [reply src/components/NcDialogButton/NcDialogButton.vue:113]: done

**Comment by susnux** [comment]
Added an example with loading buttons.

File: src/components/NcDialogButton/NcDialogButton.vue [hunk 2]
```diff
@@ -99,16 +105,30 @@ export default defineComponent({
...
+			if (isLoading.value) {
+				return
+			}
+
+			isLoading.value = true
+			try {
+				const response = await props.callback?.()
+				if (response !== false) {
+					emit('click', e)
+				}
```
Comment (ShGKme) [inline src/components/NcDialogButton/NcDialogButton.vue:125]: nitpick: to not feel like this is always an HTTP request.
```suggestion
				const result = await props.callback?.()
				if (result !== false) {
					emit('click', e)
				}
```

Comment (ShGKme) [reply src/components/NcDialogButton/NcDialogButton.vue:125]: Maybe add it to the event payload?

```suggestion
				const response = await props.callback?.()
				if (response !== false) {
					emit('click', e, response)
				}
```

**Review by ShGKme — APPROVED** [review]
Which loading it's really awesome now

File: src/components/NcDialogButton/NcDialogButton.vue (hunk 2 above)
Comment (susnux) [reply src/components/NcDialogButton/NcDialogButton.vue:125]: For me it was not a HTTP response but the "response of the callback", but also fine with naming it result :sweat_smile:

**Review by st3iny — APPROVED** [review]

**Pull request merged**